src/main/java/org/broadinstitute/hellbender/cmdline/PicardCommandLineProgram.java

```diff
@@ -35,20 +36,36 @@
     public Object instanceMain(final String[] argv) {
         // set general SAM/BAM parameters
         SamReaderFactory.setDefaultValidationStringency(VALIDATION_STRINGENCY);
+
+        // defer to parent
+        return super.instanceMain(argv);
+    }
+
+    /**
+     * Create a common SAMFileWriter for use with Picard tools.
+     *
+     * @param outputFile    - if this file has a .cram extension then a reference is required. Can not be null.
+     * @param referenceFile - the reference source to use. Can not be null if a output file has a .cram extension.
+     * @param header        - header to be used for the output writer
+     * @param preSorted     - if true then the records must already be sorted to match the header sort order
+     * @return SAMFileWriter
+     */
+    protected SAMFileWriter createCommonSAMWriter(
+            final File outputFile,
+            final File referenceFile,
+            final SAMFileHeader header,
+            final boolean preSorted)
+    {
         BlockCompressedOutputStream.setDefaultCompressionLevel(COMPRESSION_LEVEL);
 
-        if (MAX_RECORDS_IN_RAM != null) {
-            SAMFileWriterImpl.setDefaultMaxRecordsInRam(MAX_RECORDS_IN_RAM);
-        }
```

Comment:
Think you should still be propagating these values (MAX_RECORDS_IN_RAM) to the defaults -- ie., continue to call `SAMFileWriterImpl.setDefaultMaxRecordsInRam()`, `SAMFileWriterFactory.setDefaultCreateIndexWhileWriting()`, etc. in `instanceMain()`, in addition to setting these on the factory you create within the new createSAMWriter method.